src/Microsoft.FeatureManagement.AspNetCore/DefaultHttpTargetingContextAccessor.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.AspNetCore.Http;
+using Microsoft.FeatureManagement.FeatureFilters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Provides a default implementation of <see cref="ITargetingContextAccessor"/> that creates <see cref="TargetingContext"/> using info from the current HTTP request.
+    /// </summary>
+    internal sealed class DefaultHttpTargetingContextAccessor : ITargetingContextAccessor
+    {
+        /// <summary>
+        /// The key used to store and retrieve the <see cref="TargetingContext"/> from the <see cref="HttpContext"/> items.
+        /// </summary>
+        private static object _cacheKey = new object();
+
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        /// <summary>
+        /// Creates an instance of the DefaultHttpTargetingContextAccessor
+        /// </summary>
+        public DefaultHttpTargetingContextAccessor(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor ?? throw new ArgumentNullException(nameof(httpContextAccessor));
+        }
+
+        /// <summary>
+        /// Gets <see cref="TargetingContext"/> from the current HTTP request.
+        /// </summary>
+        public ValueTask<TargetingContext> GetContextAsync()
+        {
+            HttpContext httpContext = _httpContextAccessor.HttpContext;
+
+            //
+            // Try cache lookup
+            if (httpContext.Items.TryGetValue(_cacheKey, out object value))
+            {
+                return new ValueTask<TargetingContext>((TargetingContext)value);
+            }
+
+            //
+            // Treat user identity name as user id
+            ClaimsPrincipal user = httpContext.User;
+
+            string userId = user?.Identity?.Name;
+
+            //
+            // Treat claims of type Role as groups
+            IEnumerable<string> groups = httpContext.User.Claims
+                .Where(c => c.Type == ClaimTypes.Role)
+                .Select(c => c.Value);
```

Comment:
Should we force the query evaluation here?

Reply:
Ideally no- do you have an idea for how to avoid it?

Reply:
I misunderstood this comment. I think you're right, and we should force evaluation. Adjusting this to:

```csharp
IEnumerable<string> groups = httpContext.User.Claims
    .Where(c => c.Type == ClaimTypes.Role)
    .Select(c => c.Value)
    .ToList();
```

Reply:
Sounds good.